rust/lance/src/index/vector/hnsw.rs

```diff
@@ -179,7 +180,7 @@ impl VectorIndex for HNSWIndex {
     }
 
     fn use_residual(&self) -> bool {
-        false
+        true
```

Comment:
Can we pass a flag in? iirc, IVF_HNSW does use residual, but pure `HNSW` does not, right?